ansible/deploy-openstack-config.yml

```diff
@@ -0,0 +1,184 @@
+        dest: '{{ src_directory }}/{{ kayobe_config_name }}/etc/kayobe/hooks/{{ item.dest }}'
+        state: link
+      loop:
+        - { src: growroot.yml, dest: overcloud-host-configure/pre.d/5-growroot.yml }
+        - { src: fix-hostname.yml, dest: overcloud-host-configure/pre.d/10-fix-hostname.yml }
+        - { src: fix-networking.yml, dest: overcloud-host-configure/pre.d/15-fix-networking.yml }
+        - { src: configure-vxlan.yml, dest: overcloud-host-configure/pre.d/20-configure-vxlan.yml }
+        - { src: fix-networking.yml, dest: seed-host-configure/pre.d/15-fix-networking.yml }
+        - { src: configure-vxlan.yml, dest: seed-host-configure/pre.d/20-configure-vxlan.yml }
+
+
+    - name: Ensure Admin Overcloud Network file is present
+      ansible.builtin.copy:
+        src: "files/admin-oc-networks.yml"
+        dest: "{{ src_directory }}/{{ kayobe_config_name }}/etc/kayobe/environments/{{ kayobe_config_environment }}/admin-oc-networks.yml"
+        mode: "0644"
+
+    - name: Ensure Hosts file is present
+      ansible.builtin.copy:
+        src: "files/hosts"
+        dest: "{{ src_directory }}/{{ kayobe_config_name }}/etc/kayobe/environments/{{ kayobe_config_environment }}/inventory/hosts"
+        mode: "0644"
+
+    - name: Ensure Kayobe repository is present
+      ansible.builtin.git:
+        repo: "{{ kayobe_repo }}"
+        version: "{{ kayobe_version }}"
+        dest: "{{ src_directory }}/{{ kayobe_name }}"
+        update: false
+
+    - name: Ensure `venvs` directory exists
+      ansible.builtin.file:
+        path: "{{ ansible_env.HOME }}/venvs"
+        state: directory
+        mode: "0755"
+
+    - name: Ensure `kayobe` is present
+      ansible.builtin.pip:
+        name: "{{ src_directory }}/{{ kayobe_name }}/"
+        virtualenv: "{{ ansible_env.HOME }}/venvs/kayobe"
+        state: present
+
+    - name: Ensure vault password is present
+      ansible.builtin.copy:
+        src: "{{ vault_password_path }}"
+        dest: "{{ ansible_env.HOME }}/vault-pw"
+        mode: "0600"
+
+    - name: Ensure public key is present
+      ansible.builtin.copy:
+        src: "{{ public_key_path }}"
+        dest: "{{ ansible_env.HOME }}/.ssh/id_rsa.pub"
+        mode: "0644"
+
+    - name: Ensure private key is present
+      ansible.builtin.copy:
+        src: "{{ private_key_path }}"
+        dest: "{{ ansible_env.HOME }}/.ssh/id_rsa"
+        mode: "0600"
+
+    - name: Ensure OpenStack Config repository is present
+      ansible.builtin.git:
+        repo: "{{ openstack_config_repo }}"
+        dest: "{{ src_directory }}/{{ openstack_config_name }}"
+        version: "{{ openstack_config_version }}"
+        update: false
+
+    - name: Ensure `openstack` virtualenv is present
+      ansible.builtin.pip:
+        requirements: "{{ src_directory }}/{{ openstack_config_name }}/requirements.txt"
+        virtualenv: "{{ src_directory }}/{{ openstack_config_name }}/ansible/openstack-config-venv"
```

Comment:
Typically we use `"{{ src_directory }}/{{ openstack_config_name }}/venv"`. This one might get ignored since it's in a subdirectory